File: 2048-build-array-from-permutation/2048-build-array-from-permutation.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var buildArray = function(nums) {
    let temp = [];
    for(let i=0;i<nums.length;i++){
        temp[i] = nums[nums[i]];
    }

    for(let i=0;i<nums.length;i++){
        nums[i] = temp[i];
    }
    return nums;
};

File: 2048-build-array-from-permutation/2048-build-array-from-permutation.java
class Solution {
    public int[] buildArray(int[] nums) {
        int[] ind = new int[nums.length];
        for(int i=0;i<nums.length;i++){
            ind[i] = nums[nums[i]];
        }
        int[] res = new int[nums.length];
        for(int i=0;i<ind.length;i++){
            res[i] = ind[i];
        }
        return res;
    }
}